File: android/app/src/main/kotlin/com/pulsetrack/app/FontScaleEventChannel.kt
package com.pulsetrack.app

import android.content.Context
import io.flutter.plugin.common.EventChannel

class FontScaleEventChannel(private val context: Context) : EventChannel.StreamHandler {
    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
        emit(context.resources.configuration.fontScale)
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }

    fun emit(value: Float) {
        sink?.success(value.toDouble())
    }
}

File: ios/Runner/FontScaleEventChannel.swift
import Flutter
import UIKit

final class FontScaleEventChannel: NSObject, FlutterStreamHandler {
    private var sink: FlutterEventSink?

    static func register(with controller: FlutterViewController) {
        let channel = FlutterEventChannel(name: "pulsetrack/font_scale", binaryMessenger: controller.binaryMessenger)
        let handler = FontScaleEventChannel()
        channel.setStreamHandler(handler)
    }

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        emit()
        NotificationCenter.default.addObserver(self, selector: #selector(emit), name: UIContentSizeCategory.didChangeNotification, object: nil)
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        NotificationCenter.default.removeObserver(self)
        sink = nil
        return nil
    }

    @objc private func emit() {
        let category = UIApplication.shared.preferredContentSizeCategory
        let value: Double
        switch category {
        case .extraSmall, .small, .medium, .large:
            value = 1.0
        case .extraLarge, .extraExtraLarge:
            value = 1.18
        default:
            value = 1.38
        }
        sink?(value)
    }
}
